source/simulator/solver_schemes.cc

```diff
@@ -1452,6 +1455,155 @@ namespace aspect
     nonlinear_solver_control.check(1,0.0);
     signals.post_nonlinear_solver(nonlinear_solver_control);
   }
+
+
+
+  template <int dim>
+  void Simulator<dim>::solve_stokes_adjoint ()
+  {
+
+    // This solver scheme only works with a specific material model and number of compositional fiels. Check that
+    // both is chosen correctly. TODO add Assert for material model
+    Assert(introspection.n_compositional_fields == 2,
+           ExcMessage ("This solver scheme requires two compositional fields for the density and viscosity kernels."));
```

Comment:
I'm checking this now at the beginning in the parameter file rather than within each individual file (e.g. the material model, the adjoint kernel postprocessor, and here). So I removed this here.